worlds/ff4fe/__init__.py

```diff
@@ -0,0 +1,488 @@
+            region.locations.append(new_location)
+
+        for event in events.boss_events:
+            region = self.multiworld.get_region(event.area, self.player)
+            new_location = FF4FELocation(self.player, event.name, None, region)
+            region.locations.append(new_location)
+
+        if not self.is_vanilla_game():
+            new_location = FF4FELocation(self.player, "Objectives Status", None, overworld)
+            overworld.locations.append(new_location)
+            new_location = FF4FELocation(self.player, "Objective Reward",
+                                         self.location_name_to_id["Objective Reward"], overworld)
+            overworld.locations.append(new_location)
+
+        for i in range(self.get_objective_count()):
+            new_location = FF4FELocation(self.player, f"Objective {i + 1} Status", None, overworld)
+            overworld.locations.append(new_location)
+
+    def create_item(self, item: str) -> FF4FEItem:
+        item_data: ItemData = [item_data for item_data in all_items if item_data.name == item].pop()
+        return FF4FEItem(item, item_data.classification, self.item_name_to_id[item], self.player)
+
+    def create_event(self, event: str) -> FF4FEItem:
+        return FF4FEItem(event, ItemClassification.progression, None, self.player)
+
+    def create_items(self) -> None:
+        item_pool_result = create_itempool(locations.all_locations, self)
+        item_pool = item_pool_result[0]
+        self.chosen_character = item_pool_result[1]
+        self.second_character = item_pool_result[2]
+        chosen_character_placed = False
+        second_character_placed = False
+
+        character_locations = locations.character_locations.copy()
+        character_locations.remove("Starting Character 1")
+        character_locations.remove("Starting Character 2")
+        if self.options.ConquerTheGiant:
+            character_locations.remove("Giant of Bab-il Character")
+        if self.options.NoFreeCharacters:
+            for location in locations.free_character_locations:
+                character_locations.remove(location)
+                self.get_location(location).place_locked_item(self.create_item("None"))
+                item_pool.remove("None")
+        if self.options.NoEarnedCharacters:
+            for location in locations.earned_character_locations:
+                character_locations.remove(location)
+                self.get_location(location).place_locked_item(self.create_item("None"))
+                item_pool.remove("None")
+
+        restricted_character_allow_locations = list(set(character_locations) - set(locations.restricted_character_locations)).copy()
+        restricted_character_forbid_locations = list(set(character_locations) - set(restricted_character_allow_locations)).copy()
+        self.multiworld.random.shuffle(restricted_character_allow_locations)
+        self.multiworld.random.shuffle(restricted_character_forbid_locations)
```

Comment:
```suggestion
        self.random.shuffle(restricted_character_allow_locations)
        self.random.shuffle(restricted_character_forbid_locations)
```
You should use the world's random object rather than the multiworld's. This will avoid some nondeterministic behavior.

Reply:
Fixed.